lightning-rapid-gossip-sync/src/lib.rs

```diff
@@ -68,19 +72,43 @@ pub mod error;
 /// Core functionality of this crate
 pub mod processing;
 
-/// Sync gossip data from a file
-/// Returns the last sync timestamp to be used the next time rapid sync data is queried.
+/// Rapid Gossip Sync struct
+/// See [module-level documentation] for usage.
 ///
-/// `network_graph`: The network graph to apply the updates to
-///
-/// `sync_path`: Path to the file where the gossip update data is located
-///
-pub fn sync_network_graph_with_file_path(
-	network_graph: &network_graph::NetworkGraph,
-	sync_path: &str,
-) -> Result<u32, GraphSyncError> {
-	let mut file = File::open(sync_path)?;
-	processing::update_network_graph_from_byte_stream(&network_graph, &mut file)
+/// [module-level documentation]: crate
+pub struct RapidGossipSync<NG: Deref<Target=NetworkGraph>> {
+	network_graph: NG,
+	is_initial_sync_complete: AtomicBool
```

Comment:
Is this redundant with `NetworkGraph::last_rapid_gossip_sync_timestamp`? Would be nice if `last_rapid_gossip_sync_timestamp` was in `RapidGossipSync` instead given it's not always needed with `NetworkGraph`.

Reply:
no, it's in network graph for easier serialization and persistence. Admittedly a bit of a hack, but one that users will appreciate